onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -61,33 +61,33 @@ def augment_graph(self):
         :return: augmented ONNX model
         '''
 
+        model = onnx.load(self.model_path)
+        
         added_nodes = []
         added_outputs = []
-        model = onnx.load(self.model_path)
+        to_be_calibrate_tensor = set()
+      
         for node in model.graph.node:
             should_be_calibrate = ((node.op_type in self.calibrate_op_types) and
                                 (node.name not in self.black_nodes)) or (node.name in self.white_nodes)
             if should_be_calibrate:
-                input_name = node.output[0]
-                # Adding ReduceMin nodes
-                reduce_min_name = ''
-                if node.name != '':
-                    reduce_min_name = node.name + '_ReduceMin'
-                reduce_min_node = onnx.helper.make_node('ReduceMin', [input_name], [input_name + '_ReduceMin'],
-                                                        reduce_min_name,
-                                                        keepdims=0)
-                added_nodes.append(reduce_min_node)
-                added_outputs.append(helper.make_tensor_value_info(reduce_min_node.output[0], TensorProto.FLOAT, ()))
-
-                # Adding ReduceMax nodes
-                reduce_max_name = ''
-                if node.name != '':
-                    reduce_max_name = node.name + '_ReduceMax'
-                reduce_max_node = onnx.helper.make_node('ReduceMax', [input_name], [input_name + '_ReduceMax'],
-                                                        reduce_max_name,
-                                                        keepdims=0)
-                added_nodes.append(reduce_max_node)
-                added_outputs.append(helper.make_tensor_value_info(reduce_max_node.output[0], TensorProto.FLOAT, ()))
+                to_be_calibrate_tensor.add(node.input[0])
+                to_be_calibrate_tensor.add(node.output[0])
```

Comment:
some operators to quantize have more than one input and one output, like Add. We can calibrate all the non-initializer input and all output if their type is float32. Alternatively, we can maintain a list to specify the input and output to calibrate  for each op

Reply:
A quick question, will operators like''Add'' in the future also be considered as a  calibrate_op_type?

Because when I am modifying the code, I realize that originally we are looping through the nodes and the tensors are accessed via their names.(which is by default the first attribute of node.input[0]/node.output[0]) not the actual tensor type.  So it will be difficult to check its dtype and initializer option. 

Maybe is there any way to directly loop through the tensors?